### DISCUSSION TITLE
Shop api Search query not support Currency

### DISCUSSION THREAD

**Discussion by ciocca-dev** [vendurehq/vendure#2983] [General]
Another fundamental function that is missing
I was building my shop page when I noticed that ?currencyCode=EUR doesn't work in the shop API search query
Is it possible to modify the search query to return the requested currency instead of the channel's default currency?

**Comment by thangcqUET** [comment]
The same problem

**Comment by dlhck** [comment]
Our default search plugin is not currency aware - happy to get some contributions from our community here. Have you checked our Elasticsearch search plugin or the [Advanced Search plugin](https://vendure.io/hub/vendure-plus-advanced-search-plugin)?

**Reply by thangcqUET** [reply]
@dlhck Thank you for reply. I would like to contribute this function. Not sure when I can complete it but I will try soon. Do you know another document, issues, discussion,... which is related to this function besides
- this one: [https://docs.vendure.io/reference/typescript-api/default-search-plugin/](https://docs.vendure.io/reference/typescript-api/default-search-plugin/)
- and this one: [https://github.com/vendure-ecommerce/vendure/blob/master/packages/core/src/plugin/default-search-plugin/search-strategy/postgres-search-strategy.ts#L28](https://github.com/vendure-ecommerce/vendure/blob/master/packages/core/src/plugin/default-search-plugin/search-strategy/postgres-search-strategy.ts#L28)
